### PR TITLE
refactor ABFE tests

### PR DESCRIPTION
**Pull request by xiki-tempula**
Cluster the tests based on their function.

### CODE REVIEW

File: src/alchemlyb/tests/test_workflow_ABFE.py [hunk 1]
```diff
@@ -388,62 +134,202 @@ def test_uncorr_threshold(self, workflow, monkeypatch):
...
+            workflow.estimate("aaa")
+
+    def test_single_estimator_mbar(self, workflow, monkeypatch):
+        monkeypatch.setattr(workflow, "estimator", dict())
         workflow.estimate(estimators="MBAR")
         assert len(workflow.estimator) == 1
         assert "MBAR" in workflow.estimator
         summary = workflow.generate_result()
-        assert np.isclose(summary["MBAR"]["Stages"]["TOTAL"], 2.946, 0.1)
+        assert np.isclose(summary["MBAR"]["Stages"]["TOTAL"], 21.645742066696315, 0.1)
```
Comment (orbeckst) [inline src/alchemlyb/tests/test_workflow_ABFE.py:196]: Why did the result change?

File: src/alchemlyb/tests/test_workflow_ABFE.py [hunk 2]
```diff
@@ -388,62 +134,202 @@ def test_uncorr_threshold(self, workflow, monkeypatch):
...
 
-    def test_single_estimator_ti(self, workflow):
+    def test_single_estimator_ti(self, workflow, monkeypatch):
+        monkeypatch.setattr(workflow, "estimator", dict())
+        monkeypatch.setattr(workflow, "summary", None)
         workflow.estimate(estimators="TI")
         summary = workflow.generate_result()
-        assert np.isclose(summary["TI"]["Stages"]["TOTAL"], 2.946, 0.1)
+        assert np.isclose(summary["TI"]["Stages"]["TOTAL"], 21.51472826028906, 0.1)
```
Comment (orbeckst) [inline src/alchemlyb/tests/test_workflow_ABFE.py:203]: Why result change? Different input data?

At least TI and MBAR are agreeing...

File: src/alchemlyb/tests/test_workflow_ABFE.py [hunk 3]
```diff
@@ -388,62 +134,202 @@ def test_uncorr_threshold(self, workflow, monkeypatch):
...
+        assert workflow.update_units() is None
+
+
 class Test_automatic_amber:
     """Test the full automatic workflow for load_ABFE from alchemtest.amber for
     three stage transformation."""
 
     @staticmethod
-    @pytest.fixture(scope="session")
+    @pytest.fixture(scope="class")
```
Comment (orbeckst) [inline src/alchemlyb/tests/test_workflow_ABFE.py:332]: definitely safer!

**Review by orbeckst — APPROVED** [review]
Looks much cleaner — no complaints. 

I noticed that one reference value changed but you know what you're doing so I'll merge. Just let me know what changed.

File: src/alchemlyb/tests/test_workflow_ABFE.py (hunk 1 above)
Comment (xiki-tempula) [reply src/alchemlyb/tests/test_workflow_ABFE.py:196]: Originally, I use the gmx ABFE dataset to test something then I use the gmx benzene dataset to test the numerical accuracy. There is no reason for that and it is just that I happen to add the test under the gmx benzene dataset. Now, the whole test suite is tested using the gmx ABFE dataset and hence the change in number.

**Pull request merged**